src/Http/Livewire/LivewireDatatable.php

```diff
@@ -627,45 +649,90 @@ public function refreshLivewireDatatable()
         $this->page = 1;
     }
 
+    public function getColumnDirection(string $sortString): string
+    {
+        $direction = 'desc';
+        if (Str::contains($sortString, '|')) {
+            $direction = Str::after($sortString, '|');
+            if ($direction !== 'asc' && $direction !== 'desc') {
+                throw new \Exception("Invalid direction $direction given in sanitizeColumnSort() method. Allowed values: asc, desc.");
+            }
+        }
+
+        return $direction;
+    }
+
     /**
      * Order the table by a given column index starting from 0.
      *
-     * @param  int  $index  which column to sort by
-     * @param  string|null  $direction  needs to be 'asc' or 'desc'. set to null to toggle the current direction.
+     * @param int $index which column to sort by
+     * @param string|null $direction needs to be 'asc' or 'desc'. set to null to toggle the current direction.
      * @return void
      */
     public function sort($index, $direction = null)
     {
-        if (! in_array($direction, [null, 'asc', 'desc'])) {
+        if (!in_array($direction, self::ORDER_BY_DIRECTION_STATES)) {
             throw new \Exception("Invalid direction $direction given in sort() method. Allowed values: asc, desc.");
         }
+        $key = Str::snake(Str::afterLast(get_called_class(), '\\'));
+
+        if ($this->multisortable) {
+            if (($valueIndexInSort = $this->getIndexesFromSort($this->sort, $index))->isEmpty()) {
+                if ($direction === null) {
+                    $sort = $index . '|' . $this->getColumnDirection($index);
+                } else {
+                    $sort = $index . '|' . $direction;
+                }
+                array_unshift($this->sort, $sort);
+            } else {
+                $sortIndex = $valueIndexInSort->keys()->first();
+                if ($direction === null) {
+                    $toggledDirection = $this->toggleMultisortableDirection($this->getColumnDirection($this->sort[$sortIndex]));
+                    unset($this->sort[$sortIndex]);
+                    $sort = $index . '|' . $toggledDirection;
+                    if ($toggledDirection === null) {
+                        return;
+                    }
+                    array_unshift($this->sort, $sort);
+                } else {
+                    $this->sort[$sortIndex] =
+                        $index . '|' . $direction;
+                }
+            }
+
+            $this->page = 1;
+            session()->put([$key . $this->name . '_multisort' => implode(',', $this->sort)]);
```

Comment:
This could go in a private void method.

Reply:
@michalisantoniou6 Do you mean this session()->put([$key . $this->name . '_multisort' => implode(',', $this->sort)]);